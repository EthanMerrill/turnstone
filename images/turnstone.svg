<svg stroke-miterlimit="10" viewBox="0 0 1860.44 496.158" xmlns="http://www.w3.org/2000/svg" xmlns:vectornator="http://vectornator.io" clip-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"><defs><clipPath id="prefix__a"><path d="M0 11.197h1914.99V319.75H0z"/></clipPath><clipPath id="prefix__b"><path d="M429.161 359.087h1429.59v138.377H429.161z"/></clipPath></defs><g vectornator:layerName="Untitled" fill-rule="evenodd"><g fill="#14b9de"><path clip-path="url(#prefix__a)" d="M69 253.197h40.5v-161.4h63.6v-36.6H5.7v36.6H69v161.4zM198 171.297c0 52.5 35.4 85.5 84 85.5 48 0 84-33 84-85.5v-116.1h-40.5v119.1c0 28.8-17.4 45-43.5 45-25.5 0-43.5-16.2-43.5-45v-119.1H198v116.1zM576.9 253.197l-56.1-77.7c21.6-10.8 34.8-31.2 34.8-56.4 0-38.1-30-63.9-71.4-63.9h-74.4v198h40.5v-69.9H480l51.3 69.9h45.6zm-126.6-162.3h33.9c18.3 0 30 11.4 30 28.2 0 17.1-11.1 28.5-30 28.5h-33.9v-56.7zM729.9 55.197v118.2L615 51.597h-13.2v201.6h40.5v-118.8l114.6 122.4h13.5v-201.6h-40.5zM801.9 212.997c15.3 26.7 42.9 43.8 76.5 43.8 39.6 0 69.3-23.1 69.3-59.4 0-26.7-15.6-43.2-48.9-57.9l-22.5-9.9c-16.8-7.5-23.1-12.3-23.1-22.8 0-12.3 10.5-19.2 24.6-19.2 15 0 27.6 7.5 37.8 22.8l27.6-21.3c-12.3-22.8-35.1-37.5-65.4-37.5-37.5 0-65.7 22.2-65.7 56.1 0 26.4 17.7 43.5 48 57l21.3 9.6c18.9 8.7 25.2 12.9 25.2 24.6 0 13.5-10.5 21.9-28.2 21.9-21.6 0-35.7-11.1-48.6-28.8l-27.9 21zM1028.1 253.197h40.5v-161.4h63.6v-36.6H964.8v36.6h63.3v161.4zM1623 55.197v118.2l-114.9-121.8h-13.2v201.6h40.5v-118.8l114.6 122.4h13.5v-201.6H1623zM1708.8 253.197h136.8v-36.6h-96.3v-50.1h68.1v-34.5h-68.1v-40.2h92.7v-36.6h-133.2v198z"/></g><path d="M1310.85 0c-87.35 0-158.16 70.806-158.16 158.163 0 87.345 70.81 158.156 158.16 158.156s158.16-70.811 158.16-158.156C1469.01 70.805 1398.2 0 1310.85 0z" fill="#a1cdd8"/><path d="M1311.05 189.933c-53.34 0-96.58 14.094-96.58 37.357 0 23.263 43.24 26.711 96.58 26.711 53.33 0 96.57-3.448 96.57-26.711 0-31.114-43.24-37.357-96.57-37.357z" fill="#828282"/><path d="M1303.37 199.04c-40.24-7.044-77.39 13.102-80.46 30.653-3.07 17.551 120.7-23.61 80.46-30.653z" fill="#929292"/><path d="M1315.96 132.919c-45.22 0-81.88 10.513-81.88 32.657 0 22.143 36.66 28.329 81.88 28.329 45.23 0 81.89.508 81.89-21.635 0-22.145-36.66-39.351-81.89-39.351z" fill="#b5b5b5"/><path d="M1309.45 141.587c-34.12-6.705-65.62 12.472-68.22 29.179-2.61 16.707 102.35-22.474 68.22-29.179z" fill="#c4c4c4"/><path d="M1305.38 86.938c-34.68 0-62.79 4.784-62.79 24.131s28.11 29.151 62.79 29.151 62.79-2.868 62.79-22.214c0-27.019-28.11-31.068-62.79-31.068z" fill="#828282"/><path d="M1300.39 94.512c-26.16-5.858-50.32 10.895-52.31 25.492-2 14.596 78.48-19.635 52.31-25.492z" fill="#929292"/><path d="M1318.77 54.275c-23.58 0-42.71 4.199-42.71 21.13 0 13.159 19.13 15.11 42.71 15.11 23.59 0 42.71-6.451 42.71-19.61 0-13.159-19.12-16.63-42.71-16.63z" fill="#b5b5b5"/><path d="M1315.38 59.426c-17.8-3.985-34.23 7.411-35.58 17.338-1.36 9.928 53.37-13.354 35.58-17.338z" fill="#c3c3c3"/><g fill="#828282"><path clip-path="url(#prefix__b)" d="M594.153 440.087h14.5l-29.4-67.2h-4.5l-29.5 67.2h14.5l3.7-8.6h27l3.7 8.6zm-8.4-19.7h-17.6l8.8-22.3 8.8 22.3zM701.353 440.087l-18.7-25.9c7.2-3.6 11.6-10.4 11.6-18.8 0-12.7-10-21.3-23.8-21.3h-24.8v66h13.5v-23.3h9.9l17.1 23.3h15.2zm-42.2-54.1h11.3c6.1 0 10 3.8 10 9.4 0 5.7-3.7 9.5-10 9.5h-11.3v-18.9zM754.253 419.887c.6-2.5.9-5 .9-7.4 0-12.9-9.6-23-23.3-23-14.3-.3-26.2 11.6-25.9 25.9-.3 14.3 11.8 26.2 27.1 25.9 7.3 0 13.9-2.5 19.8-7.4l-5.6-9.2c-4.3 3.3-8.9 5-13.8 5-7.2 0-12.3-3.5-13.9-9.8h34.7zm-34.7-9.5c1.5-6.1 6.1-10 11.8-10 5.9 0 10.6 3.9 10.7 10h-22.5zM807.153 410.687c0-13.7-8.3-21.2-21.2-21.2-8.7 0-15.9 3.4-21.6 10.2l7 8.3c4.4-4.7 8-6.8 13.1-6.8 6 0 9.2 3.2 9.2 8.7v2.2h-12.4c-11.5 0-18.7 5.5-18.7 14.4 0 8.7 6.6 14.8 17.1 14.8 6.3 0 11.5-2.3 15.1-6v4.8h12.4v-29.4zm-13.5 10.9c0 5.7-4.5 9.8-10.4 9.8-4.3 0-7.1-2.1-7.1-5.4 0-3.3 2.5-5 6.9-5h10.6v.6zM817.353 415.287c-.2 14.3 12 26.3 26.5 26 9.6 0 17.2-4.7 21.5-12.3l-9.4-7.2c-3.2 4.5-6.8 7.1-11.9 7.1-7.6 0-13-5.7-13-13.6 0-7.7 5.4-13.5 13-13.5 5.2 0 8.8 2.7 11.9 7.2l9.4-7.3c-4.3-7.6-11.8-12.2-21.5-12.2-14.7 0-26.5 11.3-26.5 25.8zM907.153 428.787c-3.5.3-5.6.4-6.4.4-6.4 0-9.2-2.1-9.2-7.7v-19.5h14v-11.4h-14v-12.5h-13.5v12.5h-7.8v11.4h7.8v20.7c0 12.4 7.3 18.6 19.8 18.6 3 0 6.1-.4 9.3-1.2v-11.3zM941.253 426.687c5.1 8.9 14.3 14.6 25.5 14.6 13.2 0 23.1-7.7 23.1-19.8 0-8.9-5.2-14.4-16.3-19.3l-7.5-3.3c-5.6-2.5-7.7-4.1-7.7-7.6 0-4.1 3.5-6.4 8.2-6.4 5 0 9.2 2.5 12.6 7.6l9.2-7.1c-4.1-7.6-11.7-12.5-21.8-12.5-12.5 0-21.9 7.4-21.9 18.7 0 8.8 5.9 14.5 16 19l7.1 3.2c6.3 2.9 8.4 4.3 8.4 8.2 0 4.5-3.5 7.3-9.4 7.3-7.2 0-11.9-3.7-16.2-9.6l-9.3 7zM1046.35 419.887c.6-2.5.9-5 .9-7.4 0-12.9-9.6-23-23.3-23-14.3-.3-26.197 11.6-25.897 25.9-.3 14.3 11.797 26.2 27.097 25.9 7.3 0 13.9-2.5 19.8-7.4l-5.6-9.2c-4.3 3.3-8.9 5-13.8 5-7.2 0-12.3-3.5-13.9-9.8h34.7zm-34.7-9.5c1.5-6.1 6.1-10 11.8-10 5.9 0 10.6 3.9 10.7 10h-22.5zM1099.25 410.687c0-13.7-8.3-21.2-21.2-21.2-8.7 0-15.9 3.4-21.6 10.2l7 8.3c4.4-4.7 8-6.8 13.1-6.8 6 0 9.2 3.2 9.2 8.7v2.2h-12.4c-11.5 0-18.7 5.5-18.7 14.4 0 8.7 6.6 14.8 17.1 14.8 6.3 0 11.5-2.3 15.1-6v4.8h12.4v-29.4zm-13.5 10.9c0 5.7-4.5 9.8-10.4 9.8-4.3 0-7.1-2.1-7.1-5.4 0-3.3 2.5-5 6.9-5h10.6v.6zM1142.65 390.587c-1.3-.4-2.7-.6-4.4-.6-5.5 0-10.3 2.4-13.6 6.5v-5.9h-12.4v49.5h13.5v-24.7c0-8.5 5.2-12.9 13.3-12.9h3.6v-11.9zM1148.25 415.287c-.2 14.3 12 26.3 26.5 26 9.6 0 17.2-4.7 21.5-12.3l-9.4-7.2c-3.2 4.5-6.8 7.1-11.9 7.1-7.6 0-13-5.7-13-13.6 0-7.7 5.4-13.5 13-13.5 5.2 0 8.8 2.7 11.9 7.2l9.4-7.3c-4.3-7.6-11.8-12.2-21.5-12.2-14.7 0-26.5 11.3-26.5 25.8zM1251.75 409.387c0-12-7.2-19.9-18.4-19.9-5.5 0-10.3 2-13.6 5.3v-29.7h-13.5v75h13.5v-28.8c0-5.7 3.9-9.6 9.4-9.6s9.2 3.9 9.2 9.6v28.8h13.4v-30.7zM1289.85 407.087c-.4 18.9 15.8 34.6 35.1 34.2 12.7 0 23.5-6.1 29.2-16.4l-10.2-7.2c-4.7 7.4-11 11.1-18.9 11.1-12.3 0-21.3-9-21.3-21.6 0-12.6 9-21.8 21.3-21.8 7.9 0 14.2 3.7 18.9 11.1l10.2-7.2c-5.9-10.3-16.6-16.4-29.1-16.4-19.2-.4-35.5 15.3-35.2 34.2zM1361.95 415.287c-.3 14.1 11.9 26.2 26.4 25.9 14.5.3 27-11.8 26.7-25.9.3-14.1-12.1-26-26.6-25.8-14.5-.3-26.8 11.7-26.5 25.8zm39.3.1c0 7.7-5.5 13.4-12.8 13.4-7.5 0-12.8-5.7-12.8-13.4 0-7.9 5.3-13.6 12.8-13.6 7.3 0 12.8 5.9 12.8 13.6zM1501.85 409.387c0-12-7.1-19.9-18.7-19.9-6.8 0-12.6 2.7-16.5 7.1-3.3-4.7-8.2-7.1-14.7-7.1-5.7 0-10.7 2-14 5.5v-4.4h-12.4v49.5h13.5v-28.8c0-5.9 3.6-9.6 9.1-9.6 5.3 0 8.8 3.7 8.8 9.6v28.8h13.5v-29.1c0-5.5 3.8-9.3 9.1-9.3s8.8 3.7 8.8 9.6v28.8h13.5v-30.7zM1567.05 415.287c0-14.8-10.2-25.8-24-25.8-6.8 0-12.6 2.6-15.9 7.3v-6.2h-12.4v67.5h13.5v-22.8c3.6 3.9 8.8 6 14.8 6 13.8 0 24-11.2 24-26zm-13.7.1c0 8.1-5.6 13.6-12.8 13.6-6.9 0-12.5-4.9-12.5-13.7 0-8.9 5.7-13.6 12.5-13.6 7.3 0 12.8 5.4 12.8 13.7zM1574.85 415.287c-.3 14.1 11.9 26.2 26.4 25.9 14.5.3 27-11.8 26.7-25.9.3-14.1-12.1-26-26.6-25.8-14.5-.3-26.8 11.7-26.5 25.8zm39.3.1c0 7.7-5.5 13.4-12.8 13.4-7.5 0-12.8-5.7-12.8-13.4 0-7.9 5.3-13.6 12.8-13.6 7.3 0 12.8 5.9 12.8 13.6zM1683.95 409.387c0-12-7.1-19.9-18.6-19.9-5.9 0-11 2.1-14.5 5.6v-4.5h-12.4v49.5h13.5v-28.9c0-5.7 3.9-9.6 9.4-9.6s9.2 3.9 9.2 9.6v28.9h13.4v-30.7zM1742.45 419.887c.6-2.5.9-5 .9-7.4 0-12.9-9.6-23-23.3-23-14.3-.3-26.2 11.6-25.9 25.9-.3 14.3 11.8 26.2 27.1 25.9 7.3 0 13.9-2.5 19.8-7.4l-5.6-9.2c-4.3 3.3-8.9 5-13.8 5-7.2 0-12.3-3.5-13.9-9.8h34.7zm-34.7-9.5c1.5-6.1 6.1-10 11.8-10 5.9 0 10.6 3.9 10.7 10h-22.5zM1799.35 409.387c0-12-7.1-19.9-18.6-19.9-5.9 0-11 2.1-14.5 5.6v-4.5h-12.4v49.5h13.5v-28.9c0-5.7 3.9-9.6 9.4-9.6s9.2 3.9 9.2 9.6v28.9h13.4v-30.7zM1844.15 428.787c-3.5.3-5.6.4-6.4.4-6.4 0-9.2-2.1-9.2-7.7v-19.5h14v-11.4h-14v-12.5h-13.5v12.5h-7.8v11.4h7.8v20.7c0 12.4 7.3 18.6 19.8 18.6 3 0 6.1-.4 9.3-1.2v-11.3z"/></g></g></svg>