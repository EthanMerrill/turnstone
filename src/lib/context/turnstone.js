import React, { createContext, useState, useEffect } from 'react'

const TurnstoneContext = createContext()

const TurnstoneContextProvider = (props) => {
  const { children, splitChar, styles = {}, text = '' } = props
  const [queryState, setQueryState] = useState(text)
  const [highlightedState, setHighlightedState] = useState()
  const [selectedState, setSelectedState] = useState()
  const [customStyles] = useState(styles)
  const [splitCharState] = useState(splitChar)

  useEffect(() => setQueryState(text), [text])

  return (
    <TurnstoneContext.Provider
      value={{
        queryState,
        setQueryState,
        highlightedState,
        setHighlightedState,
        selectedState,
        setSelectedState,
        customStyles,
        splitCharState
      }}>
      {children}
    </TurnstoneContext.Provider>
  )
}

export { TurnstoneContext, TurnstoneContextProvider }
