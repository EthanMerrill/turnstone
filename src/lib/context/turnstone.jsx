import React, { createContext, useReducer, useState, useEffect } from 'react'
import reducer from '../reducers/reducer'
import {setQuery} from '../actions/actions'
import undef from '../utils/undef'

const TurnstoneContext = createContext() //TODO: Rename GlobalStateContext

const TurnstoneContextProvider = (props) => {
  const { children, splitChar, styles = {}, text = '', items = [] } = props
  const [state, dispatch] = useReducer(reducer, {
    query: text,
    items,
    highlighted: undef,
    selected: undef,
    customStyles: styles,
    splitChar
  })
  const [selectedState, setSelectedState] = useState()
  const [customStyles] = useState(styles)
  const [splitCharState] = useState(splitChar)

  useEffect(() => dispatch(setQuery(text)), [text]) // TODO: Is this needed?

  return (
    <TurnstoneContext.Provider
      value={{
        state,
        dispatch,
        selectedState,
        setSelectedState,
        customStyles,
        splitCharState
      }}>
      {children}
    </TurnstoneContext.Provider>
  )
}

export { TurnstoneContext, TurnstoneContextProvider }
