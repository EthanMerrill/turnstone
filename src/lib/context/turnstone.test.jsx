import React from 'react'
import renderer from 'react-test-renderer'

import { describe, expect, test } from 'vitest'
import { TurnstoneContextProvider, TurnstoneContext } from './turnstone.jsx'

describe('TurnstoneContextProvider', () => {
  test('TurnstoneContextProvider component renders', () => {
    const component = renderer.create(
      <TurnstoneContextProvider>Test</TurnstoneContextProvider>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('TurnstoneContext', () => {
  test('TurnstoneContext is present', () => {
    expect(TurnstoneContext && typeof TurnstoneContext === 'object').toBe(true)
  })
})