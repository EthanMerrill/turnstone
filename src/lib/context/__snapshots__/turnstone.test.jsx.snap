// Vitest Snapshot v1

exports[`TurnstoneContextProvider > TurnstoneContextProvider component renders 1`] = `"Test"`;
